src/crafting.cpp

```diff
@@ -1186,12 +1188,12 @@ void player::complete_craft( item &craft, const tripoint &loc )
                 // forget byproducts below either when you fix this.
                 //
                 // Temperature is not functional for non-foods
-                newit.set_item_temperature( 293.15 );
-                newit.reset_temp_check();
+                content_ptr->set_item_temperature( 293.15 );
+                content_ptr->reset_temp_check();
             }
         }
 
-        finalize_crafted_item( newit, get_faction() );
+        finalize_crafted_item( *content_ptr, get_faction() );
```

Comment:
This line concerns me somewhat, in that in the scenario of soup, you're finalizing the contents but not the container.

I'm not entirely sure if this is a problem, but given that ownership is set in the finalization step, I worry if it might be possible to have a can of soup with different ownership than the contents of the can.